src/routes/posts.js

```diff
@@ -58,6 +58,85 @@ const fetchPosts = async (req, res, next) => {
     }
 }
 
+// ASSIGN TAG TO AN ARTICLE
+
+const assignTagToPost = async (req, res) => {
+    try {
+        const { postId } = req.params // post id;
+        const { tagId, content, title } = req.body
+
+        const postsTags = await db.query(
+            'INSERT INTO posts_tags ("postId","tagId",content,title) SELECT $1,$2,$3,$4 WHERE NOT EXISTS (SELECT * FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2) RETURNING *',
+            [postId, tagId, content, title]
+        )
+
+        if (!postsTags.rows[0]) {
+            res.status(400).json({
+                message: 'Tag is already assigned to the post',
+            })
+        } else {
+            res.status(202).json({
+                message: 'Tag has been successfully assigned to the post',
+                tag: postsTags.rows[0],
+            })
+        }
+    } catch (error) {
+        console.error(error)
+        res.status(500).json({ message: 'Internal server error', error })
+    }
+}
+
+// GET ALL ARTICLES WITH SAME TAG
+
+const queryPosts = async (req, res) => {
+    try {
+        const { tag } = req.query
+
+        // get all posts' id having the same tag
+        const postsIds = await db.query(
+            'SELECT ("postId") FROM posts_tags WHERE "tagId" = $1',
```

Comment:
Looks cool but I think we should explore a`JOIN` for this to be able to get the data in one query without making multiple trips to the database for each post.

Trying to dig up a helpful article on something related but yea lets just look around this

Reply:
seen quite some stuff, but maybe this answer can help form something like what I'm referring to https://dba.stackexchange.com/a/311188

Kindly feel free to discuss what you understand from it and what is not clear so we can collaborate on that

Reply:
I'll read up on this, just picked up postgres a few days ago so not sure if it'll take a while to understand.

Either ways I'll go through it and get back

Reply:
okay understandable. kindly do let me know if anything. any question is welcome

Reply:
hi @Yellow-May, very willing to wait on this but given you picked postgres up recently, I'm a bit worried of somehow bringing down your momentum. :)

So I also looked into this a bit and can share the query i got here so you can ask questions about it (if any) and take on any new ones moving forward. if that's okay with you

<!---```sql
SELECT p.*
FROM posts p 
INNER JOIN posts_tags pt ON p.id = pt."postId" 
WHERE pt."tagId" = $id;
```
-->

Reply:
hi @Yellow-May, very willing to wait on this but given you picked Postgres up recently, I'm a bit worried about somehow bringing down your momentum. :)

So I also looked into this a bit and can share the query i got here so you can ask questions about it (if any) and take on any new ones moving forward. if that's okay with you

Reply:
Yeah that is okay with me
I will at least learn from what you know

Reply:
okay, this is it @Yellow-May 

```sql
SELECT * FROM posts p
INNER JOIN posts_tags pt ON p.id=pt."postId"
WHERE "tagId"=$1;
```

It returns posts for the specified tag

Reply:
Thanks
I'll implement it and any issue I'll let you know